java/src/jmri/swing/NamedBeanComboBox.java

```diff
@@ -0,0 +1,511 @@
+                                                manager.getBeanTypeHandled(), bean.getFullyFormattedDisplayName()));
+                                    }
+                                }
+                            } else {
+                                if (validatingInput) {
+                                    return new Validation(Validation.Type.DANGER,
+                                            Bundle.getMessage(noMatchingBean, manager.getBeanTypeHandled(), text));
+                                }
+                            }
+                        }
+                    }
+                    return new Validation(Validation.Type.NONE, ""); // NOI18N
+                }
+            });
+        }
+        this.manager.addPropertyChangeListener("beans", managerListener);
+        this.manager.addPropertyChangeListener("DisplayListName", managerListener);
+        sort();
+        setSelectedItem(selection);
+    }
+
+    public Manager<B> getManager() {
+        return manager;
+    }
+
+    public DisplayOptions getDisplayOrder() {
+        return displayOptions;
+    }
+
+    public final void setDisplayOrder(DisplayOptions displayOrder) {
+        if (displayOptions != displayOrder) {
+            displayOptions = displayOrder;
+            sort();
+        }
+    }
+
+    /**
+     * Is this JComboBox validating typed input?
+     *
+     * @return true if validating input; false otherwise
+     */
+    public boolean isValidatingInput() {
+        return validatingInput;
+    }
+
+    /**
+     * Set if this JComboBox validates typed input.
+     *
+     * @param validatingInput true to validate; false to prevent validation
+     */
+    public void setValidatingInput(boolean validatingInput) {
+        this.validatingInput = validatingInput;
+    }
+
+    /**
+     * Is this JComboBox allowing a null object to be selected?
+     *
+     * @return true if allowing a null selection; false otherwise
+     */
+    public boolean isAllowNull() {
+        return allowNull;
+    }
+
+    /**
+     * Set if this JComboBox allows a null object to be selected. If so, the
+     * null object is placed first in the displayed list of NamedBeans.
+     *
+     * @param allowNull true if allowing a null selection; false otherwise
+     */
+    public void setAllowNull(boolean allowNull) {
+        this.allowNull = allowNull;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @return the selected item as the supported type of NamedBean or null if
+     *         there is no selection, or {@link #isAllowNull()} is true and the
+     *         null object is selected
+     */
+    @Override
+    public B getSelectedItem() {
+        return getItemAt(getSelectedIndex());
+    }
+
+    @Override
+    public void setEditable(boolean editable) {
+        if (editable && !(manager instanceof ProvidingManager)) {
+            log.error("Unable to set editable to true because not backed by editable manager");
```

Comment:
If the method setEditable() is called and the manager is not a ProvidingManager, an error is put on the log but the ComboBox is still made editable. Is this desired? For me, it would make sense to add a `return` statement after the log.error() to ensure that the combo box is not set editable if the manager is not a ProvidingManager.

Reply:
This is the desired and correct behavior. Setting a widget editable has a user interfacing impact and is used often in JMRI code for that impact even without an editable model backing the widget. I would like to change that, but it must be done in small steps, and this is one of those steps.